java-frontend/src/main/java/org/sonar/plugins/java/api/tree/Tree.java

```diff
@@ -144,9 +144,17 @@
     CASE_GROUP(CaseGroupTree.class),
 
     /**
+     * case with colon: case <expression-list> :
      * {@link CaseLabelTree}
      */
-    CASE_LABEL(CaseLabelTree.class),
+    CASE_LABEL_COLON(CaseLabelTree.class),
```

Comment:
Like mentioned earlier, I would keep only one CASE_LABEL kind and rely on a boolean.